pxtblocks/code-validation/runValidatorPlanAsync.ts

```diff
@@ -36,4 +37,13 @@ namespace pxt.blocks {
             blockResults.insufficientBlocks.length === 0;
         return success;
     }
+
+    function runValidateBlockCommentsExist(usedBlocks: Blockly.Block[], inputs: BlockCommentExistsValidatorCheck): boolean {
```

Comment:
I know you mentioned this in your PR description...but I don't think I fully understand the motivation of combining these two forms of validation into the same validator-plan-check "validator" label. I'm a little worried that having both of these utilize the same "validator" (meaning, the string we specified in the validator plan check) is going to lead to confusion down the road, especially since they're not actually using the same "validator" (meaning, the function getting called that actually runs the validation).

From a dev perspective, I worry the mismatch of "validator specified in validator plan check" and "actual validation functions" could be a point of confusion.

From a user perspective, if we imagine someone ever trying to author their own criteria, I think it's more confusing to have optional parameters that dramatically change the nature of the validation than it is to simply have two different validator choices. In this case, it's just the one parameter so it's potentially not too confusing, but as a pattern, I'd be concerned it could become untenable fairly quickly.

Reply:
Yeah, that's a good point. I'll separate them. Thinking about it a bit, it's likely that the parameterized scenario will be similar to the "only check functions" scenario, so there might be some re-use there, but like you said, just checking for any and all comments is very different.